src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,198 @@
+        break;
+      }
+      case ScanType::OpLessThan: {
+        return left < right;
+        break;
+      }
+      case ScanType::OpLessThanEquals: {
+        return left <= right;
+        break;
+      }
+      case ScanType::OpGreaterThan: {
+        return left > right;
+        break;
+      }
+      case ScanType::OpGreaterThanEquals: {
+        return left >= right;
+        break;
+      }
+    }
+    return left > right;
+  };
+
+  for (ChunkID i = ChunkID(0); i < in_table->chunk_count(); i++) {
+    const auto& chunk = in_table->get_chunk(i);
+    const auto& segment = chunk.get_segment(_column_id);
+
+    auto data_type = in_table->column_type(_column_id);
+    
+    resolve_data_type(data_type, [&] (auto type) {
+      using Type = typename decltype(type)::type;
+
+      const auto typed_value_segment = std::dynamic_pointer_cast<ValueSegment<Type>>(segment);
+      if(typed_value_segment != nullptr) {
+
+          auto values = typed_value_segment->values();
+          for (size_t index = 0; index < values.size(); index++){
+            auto value = values.at(index);
+            if(
+              comparator(
+                scan_type(),
+                value,
+                type_cast<Type>(search_value())
+              )
+            ) {
+              posList.emplace_back(RowID{i, ChunkOffset(index)});
+            };
+          }
+        }
+
+      const auto typed_dict_segment = std::dynamic_pointer_cast<DictionarySegment<Type>>(segment);
+      if (typed_dict_segment != nullptr) {
+        auto attribute_vector = typed_dict_segment->attribute_vector();
+        ValueID size = static_cast<ValueID>(attribute_vector->size());
+
+        switch (scan_type()) {
+          case ScanType::OpEquals: {
+            auto value = typed_dict_segment->find_in_dict(search_value());
+            for (auto index = ValueID(0); index < size; index++) {
+              auto attribute = typed_dict_segment->attribute_vector()->get(i);
+              if (attribute == value) {
+                posList.emplace_back(RowID({i, ChunkOffset(index)}));
+              }
+            }
+            break;
+          }
+          case ScanType::OpNotEquals: {
+            auto value = typed_dict_segment->find_in_dict(search_value());
+            for (auto index = ValueID(0); index < size; index++) {
+              auto attribute = typed_dict_segment->attribute_vector()->get(i);
+              if (attribute != value) {
+                posList.emplace_back(RowID({i, ChunkOffset(index)}));
+              }
+            }
+            break;
+          }
+          case ScanType::OpLessThan: {
+            auto value = typed_dict_segment->find_in_dict(search_value());
+            for (auto index = ValueID(0); index < size; index++) {
+              auto attribute = typed_dict_segment->attribute_vector()->get(i);
+              if (attribute < value) {
+                posList.emplace_back(RowID({i, ChunkOffset(index)}));
+              }
```

Comment:
If `find_in_dict` returns INVALID_VALUE_ID, this looks like it returns true for every value. Less/greater than should use lower_bound/upper_bound.